src/common/io_exerciser/JsonStructures.h

```diff
@@ -0,0 +1,321 @@
+#include <memory>
```

Comment:
A basic question here: why invent our own JSON structures? there are stable C++ libraries for JSON data manipulation (there is boost::json, which I do not have first hand experience with; there is Nlohmann's (which
I did use a few times), and there are others. Is there a reason to invent our own?

Reply:
BTW - Nlohmann is used by Jaegertracing, which is (for now?) part of our codebase.

Reply:
I was using the json_spirit library which we use elsewhere in Ceph and was just trying to encapsulate code which I kept repeating into one place rather than it being spread throughout ceph_test_rados_io_sequence and duplicated in other classes, such as decoding the strings into the various types of the structures I wanted to use. 

It was originally meant to be more a class for just storing structs that I would decode the json strings into, but looking back I've slowly absorbed the json decoding logic of json_spirit into it.

I'm happy to change this over to Nlohmann json which might match the desired interface I was after with the structs better anyway if we want, as building a new JSON library wasn't my intention here.